lib/brainstem/dsl/configuration.rb

```diff
@@ -11,8 +11,13 @@ class Configuration
       # @params [Object] parent_configuration The parent configuration object
       #   which the new configuration object should use as a base.
       def initialize(parent_configuration = nil)
-        @parent_configuration = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
-        @storage = ActiveSupport::HashWithIndifferentAccess.new
+        @parent_configuration  = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
+        @storage               = ActiveSupport::HashWithIndifferentAccess.new
+
+        self.nonheritable_keys =
+          parent_configuration.respond_to?(:nonheritable_keys) &&
+          parent_configuration.nonheritable_keys ||
```

Comment:
So we *inherit* our parent's non-inheritable key list?  If the parent then changes this list, do we change?